lms/djangoapps/ccx/tests/test_views.py

```diff
@@ -394,6 +394,10 @@ def test_create_ccx(self, ccx_name='New CCX'):
         course_enrollments = get_override_for_ccx(ccx, self.course, 'max_student_enrollments_allowed')
         self.assertEqual(course_enrollments, settings.CCX_MAX_STUDENTS_ALLOWED)
 
+        # check if the course display name is properly set
+        course_display_name = get_override_for_ccx(ccx, self.course, 'display_name')
+        self.assertEqual(course_display_name, ccx_name)
```

Comment:
This is fine to test, but the interface we're really interested in is the modulestore. Please check that when you ask for CCX Course's root node from the modulestore and ask for its display name, it gives you the overridden name and not the name of the root course.

Reply:
Hi @ormsbee  I checked for a particular course on my local devstack and found out
that `display_name` is not getting stored in the modulestore. Infact none of the
fields(start, due, max_student_enrollments_allowed) is getting saved/updated.
Here is the code snippet which I used to verify.

```python
>>> from xmodule.modulestore.django import modulestore
>>> store = modulestore()

>>> ccx = CustomCourseForEdX.objects.get()
>>> ccx.display_name
Economics Introduction Child Course
>>> ccxm = store.get_course(ccx.course_id)
>>> ccxm.display_name
Introduction to Economics Parent Course

>>> ccx.start
datetime.datetime(2018, 8, 27, 6, 22, 43, tzinfo=tzutc())
>>> ccxm.start
datetime.datetime(2030, 1, 1, 0, 0, tzinfo=<bson.tz_util.FixedOffset object at 0x7f9bc42ae9d0>)
```

We can see that `ccx` has the right start date, where as course object from `modulestore` has the
default date.

I did some more findings and figured that `grading_policy` is getting stored via `override_field_for_ccx`.
I tried updating it, but could not see the reflected changes

```python
# assume that `new_grading_policy` variables holds a proper json object but different from `ccxm.grading_policy`
>>> from ccx_keys.locator import CCXLocator
>>> from xmodule.modulestore.django import SignalHandler
>>> from lms.djangoapps.ccx.overrides import override_field_for_ccx
>>> from courseware.courses import get_course_by_id

>>> course = get_course_by_id(ccx.course_id, depth=None)
>>> override_field_for_ccx(ccx, course, 'grading_policy', new_grading_policy)
>>> ccxm.grading_policy
# still the same object

# even after re-fetching the object from `modulestore`
>>> ccxm = store.get_course(ccx.course_id)
>>> ccxm.grading_policy
```

If I write the test, it will fail because the modulestore wont give the updated name.

CC @itsjeyd

Reply:
The `course_id` attribute of the `CustomCourseForEdX` model is the `course_id` of the root course. But if you do:

```python
ccx = CustomCourseForEdX.objects.get()
ccxm = store.get_course(ccx.locator)
```

Then `ccxm` should be pulling the CCX course with overrides from the modulestore and should hopefully have the correct `display_name`.

Reply:
@ormsbee : I have added [test](https://github.com/edx/edx-platform/pull/18808/files#diff-1f353962b6fa14872960bdbb72548764R401) for modulestore, but it is still giving the [old course name](https://build.testeng.edx.org/job/edx-platform-python-unittests-pr/56133/testReport/junit/lms.djangoapps.ccx.tests.test_views/TestCoachDashboard/test_create_ccx/)